app/src/main/java/com/example/solanakotlincomposescaffold/usecase/AccountBalanceUseCase.kt

```diff
@@ -0,0 +1,45 @@
+package com.example.solanakotlincomposescaffold.usecase
+
+import android.net.Uri
+import android.util.Log
+import com.example.solanakotlincomposescaffold.networking.KtorHttpDriver
+import com.funkatronics.encoders.Base58
+import com.funkatronics.networking.Rpc20Driver
+import com.funkatronics.rpccore.JsonRpc20Request
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.add
+import kotlinx.serialization.json.buildJsonArray
+import java.util.UUID
+
+object AccountBalanceUseCase {
+    private val TAG = AccountBalanceUseCase::class.simpleName
+    suspend operator fun invoke(rpcUri: Uri, address: ByteArray): Long =
+        withContext(Dispatchers.IO) {
+            val rpc = Rpc20Driver(rpcUri.toString(), KtorHttpDriver())
+            val requestId = UUID.randomUUID().toString()
+            val request = createBalanceRequest(address, requestId)
+            val response = rpc.makeRequest(request, BalanceResponse.serializer())
```

Comment:
What happens if the response doesn't match expected `BalanceResponse.serializer()` format? Does it hit that error on line 25?

Reply:
this [gets caught in the RPC driver](https://github.com/Funkatronics/RpcCore/blob/40687d9debe87345162bd5d4a8d8c3c5cffbe9b8/rpccore/src/commonMain/kotlin/com/funkatronics/networking/Rpc20Driver.kt#L24C18-L24C18) and an error is returned.